blockchain/error.go

```diff
@@ -71,574 +70,461 @@ func (e AssertError) Error() string {
-	ErrBadNumPayees:           "ErrBadNumPayees",
-	ErrBadPayeeScriptVersion:  "ErrBadPayeeScriptVersion",
-	ErrBadPayeeScriptType:     "ErrBadPayeeScriptType",
-	ErrMismatchedPayeeHash:    "ErrMismatchedPayeeHash",
-	ErrBadPayeeValue:          "ErrBadPayeeValue",
-	ErrSSGenSubsidy:           "ErrSSGenSubsidy",
-	ErrImmatureTicketSpend:    "ErrImmatureTicketSpend",
-	ErrTicketInputScript:      "ErrTicketInputScript",
-	ErrInvalidRevokeInput:     "ErrInvalidRevokeInput",
-	ErrSSRtxPayees:            "ErrSSRtxPayees",
-	ErrTxSStxOutSpend:         "ErrTxSStxOutSpend",
-	ErrRegTxCreateStakeOut:    "ErrRegTxCreateStakeOut",
-	ErrInvalidFinalState:      "ErrInvalidFinalState",
-	ErrPoolSize:               "ErrPoolSize",
-	ErrForceReorgWrongChain:   "ErrForceReorgWrongChain",
-	ErrForceReorgMissingChild: "ErrForceReorgMissingChild",
-	ErrBadStakebaseValue:      "ErrBadStakebaseValue",
-	ErrDiscordantTxTree:       "ErrDiscordantTxTree",
-	ErrStakeFees:              "ErrStakeFees",
-	ErrNoStakeTx:              "ErrNoStakeTx",
-	ErrBadBlockHeight:         "ErrBadBlockHeight",
-	ErrBlockOneTx:             "ErrBlockOneTx",
-	ErrBlockOneInputs:         "ErrBlockOneInputs",
-	ErrBlockOneOutputs:        "ErrBlockOneOutputs",
-	ErrNoTax:                  "ErrNoTax",
-	ErrExpiredTx:              "ErrExpiredTx",
-	ErrExpiryTxSpentEarly:     "ErrExpiryTxSpentEarly",
-	ErrFraudAmountIn:          "ErrFraudAmountIn",
-	ErrFraudBlockHeight:       "ErrFraudBlockHeight",
-	ErrFraudBlockIndex:        "ErrFraudBlockIndex",
-	ErrZeroValueOutputSpend:   "ErrZeroValueOutputSpend",
-	ErrInvalidEarlyVoteBits:   "ErrInvalidEarlyVoteBits",
-	ErrInvalidEarlyFinalState: "ErrInvalidEarlyFinalState",
-	ErrKnownInvalidBlock:      "ErrKnownInvalidBlock",
-	ErrInvalidAncestorBlock:   "ErrInvalidAncestorBlock",
-	ErrInvalidTemplateParent:  "ErrInvalidTemplateParent",
-}
-
-// String returns the ErrorCode as a human-readable name.
-func (e ErrorCode) String() string {
-	if s := errorCodeStrings[e]; s != "" {
-		return s
-	}
-	return fmt.Sprintf("Unknown ErrorCode (%d)", int(e))
+// Error satisfies the error interface and prints human-readable errors.
+func (e ErrorKind) Error() string {
+	return string(e)
 }
 
-// RuleError identifies a rule violation.  It is used to indicate that
-// processing of a block or transaction failed due to one of the many validation
-// rules.  The caller can use type assertions to determine if a failure was
-// specifically due to a rule violation and access the ErrorCode field to
-// ascertain the specific reason for the rule violation.
+// RuleError identifies a rule violation related to blocks. It has
+// full support for errors.Is and errors.As, so the caller can ascertain the
+// specific reason for the error by checking the underlying error.
 type RuleError struct {
-	ErrorCode   ErrorCode // Describes the kind of error
-	Description string    // Human readable description of the issue
+	Description string
```

Comment:
Not a big deal, but all the others put the error first.  Better to be consistent imo.